### PR TITLE
Added InternalTestFilter

### PR DESCRIPTION
**Pull request by mathieuroques**

### CODE REVIEW

File: python-client/giskard/slicing/filters.py [hunk 1]
```diff
@@ -9,6 +12,35 @@ def filter(self, slices):
         raise NotImplementedError()
 
 
+class InternalTestFilter(DataSliceFilter):
+
+    def __init__(self, model, og_dataset, test=test_diff_f1, threshold=0.1):
```
Comment (mattbit) [inline python-client/giskard/slicing/filters.py:17]: Better not to pass mutables as arguments. Better to have `test=None` and then
```py
self.test = test if test is not None else test_diff_f1
```

Comment (mattbit) [inline python-client/giskard/slicing/filters.py:17]: `og_dataset` → `dataset`?

File: python-client/giskard/slicing/filters.py [hunk 2]
```diff
@@ -9,6 +12,35 @@ def filter(self, slices):
...
+        self.dataset = og_dataset
+        self.model = model
+        self.test = test
+        self.threshold = threshold
+
+    def _diff_test(self, data_slice):
+        # Convert slice to Giskard Dataframe
+        g_dataset = Dataset(data_slice,
```
Comment (mattbit) [inline python-client/giskard/slicing/filters.py:25]: What is `data_slice`? I don’t think we can build `Dataset` from that?

File: python-client/giskard/slicing/filters.py [hunk 3]
```diff
@@ -9,6 +12,35 @@ def filter(self, slices):
...
+                            target_col=self.dataset.target,
+                            class_labels=self.dataset.class_labels,
+                            column_types=self.dataset.column_types
+                            )
+        # Apply the test
+        test_res = self.test(
+            actual_slice=g_dataset,
+            reference_slice=self.dataset,
```
Comment (mattbit) [inline python-client/giskard/slicing/filters.py:33]: Would prefer to have exclude the slice from the reference, for independence.

Comment (mathieuroques) [reply python-client/giskard/slicing/filters.py:33]: Noted

File: python-client/giskard/slicing/filters.py [hunk 4]
```diff
@@ -1,6 +1,9 @@
 import logging
 from scipy import stats
 
+from giskard.data.dataset import Dataset
```
[the remark this answers is not in the document]
Comment (mathieuroques) [reply python-client/giskard/slicing/filters.py:4]: Need to change:
- merge dataset and model in the same input

File: python-client/giskard/scanner/performance.py [hunk 5]
```diff
@@ -16,36 +16,80 @@
...
+        for s in slices:
+            actual_slices_size, reference_slices_size, metric, passed = self._diff_test(s, test, threshold)
+            if not passed:
+                issues.append(Issue(s, metric, actual_slices_size, reference_slices_size, model, dataset))
+
+        return issues
+
+    def _diff_test(self, data_slice, test, threshold):
+        # Convert slice to Giskard Dataframe
+        slice_dataset = Dataset(data_slice.data)
```
Comment (mattbit) [inline python-client/giskard/scanner/performance.py:79]:
```py
sliced_dataset = self.dataset.slice(slice_fn)
```

Comment (mattbit) [reply python-client/giskard/scanner/performance.py:79]: `self.model` and `self.dataset` might be undefined, you should pass explicitly to `_diff_test`

File: python-client/giskard/scanner/performance.py [hunk 6]
```diff
@@ -16,36 +16,80 @@
...
         self.model = model
         self.dataset = dataset
+        self.test_names = test_names
 
-    def run(self, model: Optional[BaseModel] = None, dataset: Optional[Dataset] = None, slicer="opt"):
+    def run(self, model: Optional[BaseModel] = None, dataset: Optional[Dataset] = None, slicer="opt", test_names=None,
+            threshold=0.1):
         model = model or self.model
         dataset = dataset or self.dataset
+        test_names = test_names or self.test_names
```
Comment (mattbit) [inline python-client/giskard/scanner/performance.py:36]: Could we add support for multiple tests? E.g. `tests=["f1", "accuracy"]`.

File: python-client/giskard/scanner/performance.py [hunk 7]
```diff
@@ -16,36 +16,80 @@
...
+        issues = self._find_issues(slices, model, dataset, test, threshold)
 
         return PerformanceScanResult(issues)
 
+    def _find_issues(self, slices, model, dataset, test, threshold):
+        issues = []
+        for s in slices:
+            actual_slices_size, reference_slices_size, metric, passed = self._diff_test(s, test, threshold)
+            if not passed:
+                issues.append(Issue(s, metric, actual_slices_size, reference_slices_size, model, dataset))
```
Comment (mattbit) [inline python-client/giskard/scanner/performance.py:73]: Issue takes `model`, `dataset`, and `test_result`

**Review by mattbit — CHANGES_REQUESTED** [review]

File: python-client/giskard/scanner/performance.py [hunk 8]
```diff
@@ -61,30 +63,31 @@ def run(self, model: Optional[BaseModel] = None, dataset: Optional[Dataset] = No
...
+                if not test_result.passed:
+                    issues.append(Issue(s, model, dataset, test_result))
 
         return issues
 
-    def _diff_test(self, data_slice, test, threshold):
+    def _diff_test(self, slice_fn, model, dataset, test, threshold):
         # Convert slice to Giskard Dataframe
-        slice_dataset = Dataset(data_slice.data)
+        sliced_dataset = self.dataset.slice(slice_fn)
```
Comment (mattbit) [inline python-client/giskard/scanner/performance.py:82]: `dataset`

**Review by mattbit — CHANGES_REQUESTED** [review]
All good except for a typo! 🔥

**Review by mattbit — APPROVED** [review]

**Pull request merged**